### PR TITLE
Fix #221

### LINKED ISSUE #221: make all objects non-rej by default, and only UCODE_SMALL enables rej
this will prevent user confusion, and makes enabling rej completely intentional

note: OBJECTS_REJ should be disabled by default

### PR DESCRIPTION
**Pull request by FazanaJ**
Makes Rejection microcode an opt in for objects, rather than an opt out.

### CODE REVIEW

File: data/behavior_data.c [hunk 1]
```diff
@@ -1457,7 +1457,7 @@ const BehaviorScript bhvDddMovingPole[] = {
 
 const BehaviorScript bhvBitfsTiltingInvertedPyramid[] = {
     BEGIN(OBJ_LIST_SURFACE),
-    OR_LONG(oFlags, (OBJ_FLAG_COMPUTE_DIST_TO_MARIO | OBJ_FLAG_SET_FACE_YAW_TO_MOVE_YAW | OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_UCODE_LARGE)),
+    OR_LONG(oFlags, (OBJ_FLAG_COMPUTE_DIST_TO_MARIO | OBJ_FLAG_SET_FACE_YAW_TO_MOVE_YAW | OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_UCODE_SMALL)),
```
Comment (Mr-Wiseguy) [inline data/behavior_data.c]: Behavior was inverted here.

File: include/config/config_graphics.h [hunk 2]
```diff
@@ -30,9 +30,9 @@
 #define IA8_30FPS_COINS
 
 // Use .rej microcode for certain objects (experimental - only should be used when F3DZEX_GBI_2 is defined).
-// So far - recent tests have show that this has reduced performance overall due to CPU load,
-// though in some cases it may perform better.
-// #define OBJECTS_REJ
+// For advanced users only. Does not work perfectly out the box, best used when exported actor models are
+// using 64 vertex sizes, offered by Fast64 in the microcode menu.
+ #define OBJECTS_REJ
```
Comment (thecozies) [inline include/config/config_graphics.h:35]: I don't have it in my heart to default this to being on yet

File: data/behavior_data.c [hunk 3]
```diff
@@ -1783,7 +1783,7 @@ const BehaviorScript bhvFloorSwitchHiddenObjects[] = {
 
 const BehaviorScript bhvHiddenObject[] = {
     BEGIN(OBJ_LIST_SURFACE),
-    OR_LONG(oFlags, (OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_UCODE_LARGE)),
+    OR_LONG(oFlags, (OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_UCODE_SMALL)),
```
Comment (Mr-Wiseguy) [inline data/behavior_data.c]: Behavior was inverted here.

File: data/behavior_data.c [hunk 4]
```diff
@@ -1793,7 +1793,7 @@ const BehaviorScript bhvHiddenObject[] = {
 
 const BehaviorScript bhvBreakableBox[] = {
     BEGIN(OBJ_LIST_SURFACE),
-    OR_LONG(oFlags, (OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_DONT_CALC_COLL_DIST | OBJ_FLAG_UCODE_LARGE)),
+    OR_LONG(oFlags, (OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_DONT_CALC_COLL_DIST | OBJ_FLAG_UCODE_SMALL)),
```
Comment (Mr-Wiseguy) [inline data/behavior_data.c]: Behavior was inverted here.

File: include/config/config_graphics.h (hunk 2 above)
Comment (FazanaJ) [reply include/config/config_graphics.h:35]: That's what we call in the business "an oopsie woopsie"

File: data/behavior_data.c [hunk 5]
```diff
@@ -2179,15 +2179,15 @@ const BehaviorScript bhvTankFishGroup[] = {
...
+    OR_LONG(oFlags, OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_UCODE_SMALL),
     CALL_NATIVE(bhv_checkerboard_elevator_group_init),
     DELAY(1),
     DEACTIVATE(),
 };
 
 const BehaviorScript bhvCheckerboardPlatformSub[] = {
     BEGIN(OBJ_LIST_SURFACE),
-    OR_LONG(oFlags, (OBJ_FLAG_COMPUTE_DIST_TO_MARIO | OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_UCODE_LARGE)),
+    OR_LONG(oFlags, (OBJ_FLAG_COMPUTE_DIST_TO_MARIO | OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_UCODE_SMALL)),
```
Comment (Mr-Wiseguy) [inline data/behavior_data.c]: Behavior was inverted here.

File: data/behavior_data.c [hunk 6]
```diff
@@ -2480,7 +2480,7 @@ const BehaviorScript bhvPiranhaPlant[] = {
 
 const BehaviorScript bhvLllBowserPuzzlePiece[] = {
     BEGIN(OBJ_LIST_SURFACE),
-    OR_LONG(oFlags, (OBJ_FLAG_COMPUTE_DIST_TO_MARIO | OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_DONT_CALC_COLL_DIST | OBJ_FLAG_UCODE_LARGE)),
+    OR_LONG(oFlags, (OBJ_FLAG_COMPUTE_DIST_TO_MARIO | OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_DONT_CALC_COLL_DIST | OBJ_FLAG_UCODE_SMALL)),
```
Comment (Mr-Wiseguy) [inline data/behavior_data.c]: Behavior was inverted here.

File: data/behavior_data.c [hunk 7]
```diff
@@ -3022,7 +3022,7 @@ const BehaviorScript bhvGhostHuntBoo[] = {
 
 const BehaviorScript bhvHiddenStaircaseStep[] = {
     BEGIN(OBJ_LIST_SURFACE),
-    OR_LONG(oFlags, (OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_UCODE_LARGE)),
+    OR_LONG(oFlags, (OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_UCODE_SMALL)),
```
Comment (Mr-Wiseguy) [inline data/behavior_data.c]: Behavior was inverted here.

File: data/behavior_data.c [hunk 8]
```diff
@@ -3047,7 +3047,7 @@ const BehaviorScript bhvBooStaircase[] = {
 
 const BehaviorScript bhvBbhTiltingTrapPlatform[] = {
     BEGIN(OBJ_LIST_SURFACE),
-    OR_LONG(oFlags, (OBJ_FLAG_COMPUTE_ANGLE_TO_MARIO | OBJ_FLAG_COMPUTE_DIST_TO_MARIO | OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_UCODE_LARGE)),
+    OR_LONG(oFlags, (OBJ_FLAG_COMPUTE_ANGLE_TO_MARIO | OBJ_FLAG_COMPUTE_DIST_TO_MARIO | OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_UCODE_SMALL)),
```
Comment (Mr-Wiseguy) [inline data/behavior_data.c]: Behavior was inverted here.

File: data/behavior_data.c [hunk 9]
```diff
@@ -3059,7 +3059,7 @@ const BehaviorScript bhvBbhTiltingTrapPlatform[] = {
 
 const BehaviorScript bhvHauntedBookshelf[] = {
     BEGIN(OBJ_LIST_SURFACE),
-    OR_LONG(oFlags, (OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_UCODE_LARGE)),
+    OR_LONG(oFlags, (OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_UCODE_SMALL)),
```
Comment (Mr-Wiseguy) [inline data/behavior_data.c]: Behavior was inverted here.

File: data/behavior_data.c [hunk 10]
```diff
@@ -4288,15 +4288,15 @@ const BehaviorScript bhvBobBowlingBallSpawner[] = {
...
+    OR_LONG(oFlags, OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_UCODE_SMALL),
     BEGIN_LOOP(),
         CALL_NATIVE(bhv_thi_bowling_ball_spawner_loop),
     END_LOOP(),
 };
 
 const BehaviorScript bhvRrCruiserWing[] = {
     BEGIN(OBJ_LIST_DEFAULT),
-    OR_LONG(oFlags, (OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_UCODE_LARGE)),
+    OR_LONG(oFlags, (OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_UCODE_SMALL)),
```
Comment (Mr-Wiseguy) [inline data/behavior_data.c]: Behavior was inverted here.

**Review by Mr-Wiseguy — CHANGES_REQUESTED** [review]

File: data/behavior_data.c [hunk 11]
```diff
@@ -517,7 +517,7 @@ const BehaviorScript bhvCapSwitch[] = {
 
 const BehaviorScript bhvKingBobomb[] = {
     BEGIN(OBJ_LIST_GENACTOR),
-    OR_INT(oFlags, (OBJ_FLAG_COMPUTE_ANGLE_TO_MARIO | OBJ_FLAG_ACTIVE_FROM_AFAR | OBJ_FLAG_COMPUTE_DIST_TO_MARIO | OBJ_FLAG_SET_FACE_YAW_TO_MOVE_YAW | OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE)),
+    OR_LONG(oFlags, (OBJ_FLAG_COMPUTE_ANGLE_TO_MARIO | OBJ_FLAG_ACTIVE_FROM_AFAR | OBJ_FLAG_COMPUTE_DIST_TO_MARIO | OBJ_FLAG_SET_FACE_YAW_TO_MOVE_YAW | OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_UCODE_SMALL)),
```
Comment (thecozies) [inline data/behavior_data.c]: https://user-images.githubusercontent.com/79979276/147142267-c157711d-7f8b-4ac6-a7d7-a0d62d9a21a4.mp4

File: data/behavior_data.c [hunk 12]
```diff
@@ -687,7 +687,7 @@ const BehaviorScript bhvFishGroup[] = {
 
 const BehaviorScript bhvCannon[] = {
     BEGIN(OBJ_LIST_LEVEL),
-    OR_INT(oFlags, (OBJ_FLAG_ACTIVE_FROM_AFAR | OBJ_FLAG_COMPUTE_DIST_TO_MARIO | OBJ_FLAG_SET_FACE_YAW_TO_MOVE_YAW | OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE)),
+    OR_LONG(oFlags, (OBJ_FLAG_ACTIVE_FROM_AFAR | OBJ_FLAG_COMPUTE_DIST_TO_MARIO | OBJ_FLAG_SET_FACE_YAW_TO_MOVE_YAW | OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_UCODE_SMALL)),
```
Comment (thecozies) [inline data/behavior_data.c]: https://user-images.githubusercontent.com/79979276/147142387-9fb66980-6305-4265-bd2b-92ba6bc0a99c.mp4

**Review by thecozies — CHANGES_REQUESTED** [review]
yeah i agree with wise, lets re-evaluate what should be rej in v2.1. too many of these are messed up

File: include/config/config_graphics.h (hunk 2 above)
Comment (Reonu) [reply include/config/config_graphics.h]: > I don't have it in my heart to default this to being on yet

or ever

File: data/behavior_data.c [hunk 13]
```diff
@@ -2429,7 +2429,7 @@ const BehaviorScript bhvLllTiltingInvertedPyramid[] = {
 
 const BehaviorScript bhvKoopaShell[] = {
     BEGIN(OBJ_LIST_LEVEL),
-    OR_LONG(oFlags, (OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_SILHOUETTE)), //! Silhouette doesn't show up in-game, due to combiner modes.
+    OR_INT(oFlags, (OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_SILHOUETTE)), //! Silhouette doesn't show up in-game, due to combiner modes.
```
Comment (thecozies) [inline data/behavior_data.c:2432]:
```suggestion
    OR_LONG(oFlags, (OBJ_FLAG_UPDATE_GFX_POS_AND_ANGLE | OBJ_FLAG_SILHOUETTE)), //! Silhouette doesn't show up in-game, due to combiner modes.
```

File: data/behavior_data.c [hunk 14]
```diff
@@ -3461,7 +3461,7 @@ UNUSED static const u64 behavior_data_unused_0 = 0;
 const BehaviorScript bhvMario[] = {
     BEGIN(OBJ_LIST_PLAYER),
     SET_INT(oIntangibleTimer, 0),
-    OR_LONG(oFlags, (OBJ_FLAG_PLAYER | OBJ_FLAG_SILHOUETTE)),
+    OR_INT(oFlags, (OBJ_FLAG_PLAYER | OBJ_FLAG_SILHOUETTE)),
```
Comment (thecozies) [inline data/behavior_data.c:3464]:
```suggestion
    OR_LONG(oFlags, (OBJ_FLAG_PLAYER | OBJ_FLAG_SILHOUETTE)),
```

**Review by thecozies — APPROVED** [review]
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />


the rest of my message got rejected

**Comment by FazanaJ** [comment]
boutta reject your life 🔫

**Review by Mr-Wiseguy — APPROVED** [review]
Lgtm

**Pull request merged**